kubernetes/utils/delete_from_yaml.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import re
+from os import path
+
+import yaml
+
+from kubernetes import client
+
+
+def delete_from_yaml(k8s_client, yaml_file, verbose=False,
+                     namespace="default", **kwargs):
+    """Input:
+    yaml_file: string. Contains the path to yaml file.
+    k8s_client: an ApiClient object, initialized with the client args.
+    verbose: If True, print confirmation from the create action.
+        Default is False.
+    namespace: string. Contains the namespace to create all
+        resources inside. The namespace must preexist otherwise
+        the resource creation will fail. If the API object in
+        the yaml file already contains a namespace definition
+        this parameter has no effect.
+    Available parameters for creating <kind>:
+    :param async_req bool
+    :param str pretty: If 'true', then the output is pretty printed.
+    :param str dry_run: When present, indicates that modifications
+        should not be persisted. An invalid or unrecognized dryRun
+        directive will result in an error response and no further
+        processing of the request.
+        Valid values are: - All: all dry run stages will be processed
+    Raises:
+        FailToDeleteError which holds list of `client.rest.ApiException`
+        instances for each object that failed to delete.
+    """
+    with open(path.abspath(yaml_file)) as f:
+        yml_document_all = yaml.safe_load_all(f)
+        failures = []
+        for yml_document in yml_document_all:
+            try:
+                # call delete from dict function
+                delete_from_dict(k8s_client, yml_document, verbose,
+                                 namespace=namespace, **kwargs)
+            except FailToDeleteError as failure:
+                failures.extend(failure.api_exceptions)
+        if failures:
+            raise FailToDeleteError(failures)
+
+
+def delete_from_dict(k8s_client, yml_document, verbose,
+                     namespace="default", **kwargs):
```

Comment:
add comments about what it does, Input, Returns etc

Reply:
sure will do